### PR TITLE
Update the code to Id.Web 2.5.0

### PR DESCRIPTION
**Pull request by jmprieur**
Update the daemon sample chapter 1 (call graph) to Microsoft.Identity.Web 2.0.0-preview

## Purpose
Simplify the sample by using Microsoft.Identity.Web 2.0.0-preview

### CODE REVIEW

**Comment by v-michaelmi** [comment]
LGTM

Was looking forward to seeing this! 😀

There's going to be some updates needed in some docs but once this is pushed through, I'll start integrating with the other samples 👍

**Review by v-michaelmi — APPROVED** [review]

**Comment by jmprieur** [comment]
@v-michaelmi @kalyankrishna1 yes this is more a kind of overview on how simpler the samples and quickstarts will be with the new Id.Web available on every .NET PF.
BTW if you changed the target platform from net6.0 to net472 this would work seamlessly

@mmacy I think you'll be interested too. The experience, and the docs are going to be way simpler.

**Review by jennyf19 — APPROVED** [review]
Thank you! :shipit:

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: 1-Call-MSGraph/README.md [hunk 1]
```diff
@@ -167,60 +167,72 @@ Start the application, it will display the users in the tenant.
...
+    // Get the Token acquirer factory instance. By default it reads the configuration in an appsettings.json file
+    // if it exists in the project.
+    TokenAcquirerFactory tokenAcquirerFactory = TokenAcquirerFactory.GetDefaultInstance();
+
+    // Configure the authentication options, add the services you need (Microsoft Graph, token cache)
+    IServiceCollection services = tokenAcquirerFactory.Services;
+    services.Configure<MicrosoftAuthenticationOptions>(
+              option => tokenAcquirerFactory.Configuration.GetSection("AzureAd").Bind(option))
+            .AddMicrosoftGraph()
+            .AddInMemoryTokenCaches();
```
Comment (bgavrilMS) [inline 1-Call-MSGraph/README.md:186]: The caching is not present in the sample code, just a comment.

**Review by bgavrilMS — APPROVED** [review]
Love it, LGTM

**Review by jennyf19 — APPROVED** [review]
LGTM

**Review by bgavrilMS — APPROVED** [review]

**Comment by damienbod** [comment]
Small question, why use Microsoft.Identity.Web here? Would it not be better to use GraphServiceClient direct with Azure.Identity?

**Comment by bgavrilMS** [comment]
> Small question, why use Microsoft.Identity.Web here? Would it not be better to use GraphServiceClient direct with Azure.Identity?

That's a great question. We really need to get some guidance to show differences between MSAL, Id.Web v2 and Azure SDK. 

@damienbod - Azure SDK is higher level SDK built on top of MSAL. It does not offer all the options. It does offer Managed Identity though, which we are now copying into MSAL.

**Comment by damienbod** [comment]
@bgavrilMS Thanks for your answer. I saw the managed identity integration, looks good. I believe some docs should be added that user managed identities should not be used to acquire access to Graph or Azure resources as the user managed identity ID is in this case, effectively a shared secret for any Azure resource on the same tenant. Only system managed identities are safe.

**Comment by jmprieur** [comment]
@damienbod 
We are in the process of releasing Microsoft.Identity.Web 2.x and update all the confidential client applications samples and all the docs.

**Review by jmprieur — COMMENTED** [review]
LGTM
Thanks @jennyf19

**Comment by bgavrilMS** [comment]
@jmprieur - we should discuss a migration plan from MSAL to Id.Web for this scenario. I think we may need a sample where we still showcase MSAL, as we'd like to foremost get app developers to move to MI and it's easier to jump from MSAL + cert to MSAL + MI than from MSAL + cert to Id.Web + MI. We should meet and discuss.

**Review by bgavrilMS — APPROVED** [review]

**Pull request merged**